pkg/controller/deployment/sync.go

```diff
@@ -247,14 +240,30 @@ func (dc *DeploymentController) getNewReplicaSet(deployment *extensions.Deployme
 	newRevision := strconv.FormatInt(maxOldRevision+1, 10)
 
 	existingNewRS, err := deploymentutil.FindNewReplicaSet(deployment, rsList)
```

Comment:
Before we start deep-copying replica sets, we should consolidate all annotation updates possible (revision, replicas annotations) into one.